lib/charms/mongodb/v1/mongodb_provider.py

```diff
@@ -88,6 +88,10 @@ def pass_sanity_hook_checks(self) -> bool:
         if not self.charm.db_initialised:
             return False
 
+        # This is tricky: the sanity hook checks can pass when the call is
+        # issued by a config-sever because the config-server is allowed to manage the users
+        # in MongoDB. This is not well named and does not protect integration of a config-server
+        # to a client application
```

Comment:
I believe the reason of this is that mongos charms do not have a `is_relation_feasible` attribute. So we must protect the call, by first checking what the role is.

I think the comment here should be
```
# it is important to check if the relation is feasible for mongodb charms which support a number of interfaces.
# However, the mongos charm which supports this does not need to check if the relationship is feasible
```

do you think that would improve this check? If not, would adding a function `is_relation_feasible_for_mongodb_charm` improve this bit of code?

```
def is_relation_feasible_for_mongodb_charm(self) -> bool:
         # base case, do not proceed if running on a mongos charm
         if self.charm.is_role(Config.Role.MONGOS) :
                return True
        
         return self.charm.is_relation_feasible( self.get_relation_name())
```

Reply:
The issue here is not about mongos in what I talk about in the comment: What I'm saying is that the regular hook check allows config-server to run some `mongodb_provider` code and hence the `get_relation_name` function returns a relation that does not make `is_relation_feasible` check fail. Here, at integration time (because it's a relation we're talking about and not some protected code) we need to allow only the right roles to bind on this relation. I can rephrase however a bit of the comment to make it more clear.

This will be refactored in the single kernel charm because there's an issue with split responsibilities here between what the relation does (lifecycle) and the interactions with mongodb (workload).

Reply:
thank you!